File: Payload/TikTok.app/CameraResource_i18n.bundle/bundle/Filter.bundle/Filter_F1/AmazingFeature/js/filter.js

const effect_api = "undefined" != typeof effect ? effect : "undefined" != typeof tt ? tt : "undefined" != typeof lynx ? lynx : {};
const Amaz = effect_api.getAmaz();

class filter {
    constructor() {
        this.name = "filter";
    }
    onEnable(){
        //console.log("[OnEnable]",this.name);
    }

    onStart() {
        let curJSScriptSystem = this.scene.getSystem("JSScriptSystem")
        curJSScriptSystem.clearAllEventType()
        curJSScriptSystem.addEventType(Amaz.AppEventType.COMPAT_BEF)
        
        let output_rt = this.scene.getOutputRenderTexture()
        output_rt.attachment = Amaz.RenderTextureAttachment.NONE

        this.filterMaterial = this.scene.findEntityBy("FilterEntity").getComponent("MeshRenderer").material
    }

    onUpdate(deltaTime) {

    }

    onLateUpdate(deltaTime) {
    }

    onEvent(event) {
        if (event.type == Amaz.AppEventType.COMPAT_BEF) { 
            // args: tag, path, value
            let tagName = event.args.get(0)
            
            if (tagName == "Filter_intensity"){ 
                let path = event.args.get(1)
                let value = event.args.get(2)
                if (value >= 0.0 && value <= 1.0)
                {
                   this.setFilterIntensity(value) 
                }
            }
            if (tagName == "leftSlidePosition"){
                let path = event.args.get(1)
                let value = event.args.get(2)
                //console.log("filter:SlidePositionLeft: " + value)
                this.setActiveArea(0.0, value, 0.0, 1.0)
            }
            if (tagName == "rightSlidePosition"){
                let path = event.args.get(1)
                let value = event.args.get(2)
                //console.log("filter:SlidePositionRight: " + value)
                this.setActiveArea(value, 1.0, 0.0, 1.0)
            }

        }

    }

    setFilterIntensity(intensity){
        this.filterMaterial.setFloat("intensity", intensity)
    }

    setActiveArea(leftBorder, rightBorder, topBorder, bottomBorder){
        this.filterMaterial.setVec4("activeArea", new Amaz.Vector4f(leftBorder, rightBorder, topBorder, bottomBorder))
    }
}

exports.filter = filter;

File: Payload/TikTok.app/CameraResource_i18n.bundle/bundle/Filter.bundle/Filter_F1/AmazingFeature/lua/filter.lua
local exports = exports or {}
local filter = filter or {}
filter.__index = filter

function filter.new(construct, ...)
    local self = setmetatable({}, filter)
    self.filterMaterial = nil
    return self
end

function filter:onStart(component)
    -- Amaz.LOGI("filter", "filter:onStart")

    -- disable irrelevant event
    local scriptSys = component.entity.scene:getSystem("ScriptSystem")
    scriptSys:clearAllEventType()
    scriptSys:addEventType(Amaz.BEFEventType.BET_COMPOSER)

    -- disable depth rt
    local output_rt = component.entity.scene:getOutputRenderTexture()
    output_rt.attachment = Amaz.RenderTextureAttachment.NONE

    -- find filter
    self.filterMaterial = component.entity:getComponent("MeshRenderer").material
end

function filter:onEvent(component, event)
    if event.type == Amaz.BEFEventType.BET_COMPOSER then
        if event.args:get(0) == "Filter_intensity" then
            local filterIntensity = event.args:get(2)
            if filterIntensity >= 0.0 and filterIntensity <= 1.0 then
                self:setFilterIntensity(filterIntensity)
            end
        elseif event.args:get(0) == "leftSlidePosition" then
            local leftSlidePosition = event.args:get(2)
            if leftSlidePosition == 0.0 or leftSlidePosition == 1.0 then
                self.filterMaterial:disableMacro("AE_SlidingFilter")
            else
                self.filterMaterial:enableMacro("AE_SlidingFilter", 1)
                self:setActiveArea(0.0, leftSlidePosition)
            end
        elseif event.args:get(0) == "rightSlidePosition" then
            local rightSlidePosition = event.args:get(2)
            if rightSlidePosition == 0.0 or rightSlidePosition == 1.0 then
                self.filterMaterial:disableMacro("AE_SlidingFilter")
            else
                self.filterMaterial:enableMacro("AE_SlidingFilter", 1)
                self:setActiveArea(rightSlidePosition, 1.0)
            end
        end
    end
end

function filter:setFilterIntensity(intensity)
    self.filterMaterial:setFloat("intensity", intensity)
end

function filter:setActiveArea(leftBorder, rightBorder)
    self.filterMaterial:setVec2("activeArea", Amaz.Vector2f(leftBorder, rightBorder))
end

exports.filter = filter
return exports
